Support service_account_issuer and api_audiences in alicloud_cs_kubernetes and alicloud_cs_managed_kubernetes

embik opened the pull request:
fixes #2567 

This adds the necessary fields to support of service_account_issuer and api_audiences that enables Service Account Token Volume Projection. Required a bump of the aliyungo SDK.

I'm not sure about some of the `vendor`changes to alibaba-cloud-sdk-go, I suspect that it wasn't properly updated during https://github.com/terraform-providers/terraform-provider-alicloud/commit/9d6b2611c35252f24c3a131edeca4f0cfc5d63e6

demonwy commented:
thanks for your work @embik 
it is LGTM. @xiaozhu36  please merge it.

xiaozhu36 commented on alicloud/resource_alicloud_cs_kubernetes.go:
```diff
@@ -1051,6 +1064,13 @@ func buildKubernetesArgs(d *schema.ResourceData, meta interface{}) (*cs.Delicate
+		apiAudiences = strings.Join(list, ",")
+	} else {
+		apiAudiences = ""
+	}
```
The `apiAudiences` defaults to "" and the line 1070 is needless.

embik replied:
Done!

xiaozhu36 commented on website/docs/r/cs_kubernetes.html.markdown:
```diff
@@ -96,6 +96,8 @@ The following arguments are supported:
 * `image_id` - Custom Image support. Must based on CentOS7 or AliyunLinux2.
 * `user_data` - (Optional, Available in 1.81.0+) Windows instances support batch and PowerShell scripts. If your script file is larger than 1 KB, we recommend that you upload the script to Object Storage Service (OSS) and pull it through the internal endpoint of your OSS bucket.
 * `exclude_autoscaler_nodes` - (Optional, Available in 1.88.0+) Exclude autoscaler nodes from `worker_nodes`. default: false 
+* `service_account_issuer` - (Optional, ForceNew) The issuer of the Service Account token for [Service Account Token Volume Projection](https://www.alibabacloud.com/help/doc-detail/160384.htm), corresponds to the `iss` field in the token payload. Set this to `"kubernetes.default.svc"` to enable the Token Volume Projection feature (requires specifying `api_audiences` as well).
```
Please add a tag `Available in 1.89.0+` for new parameters.

embik replied:
Done!

embik commented:
Thank you for taking a look @demonwy @xiaozhu36! I have addressed the comments and re-based on master, making the changes much cleaner. Please check again.

embik commented:
Hi @demonwy @xiaozhu36, sorry to bother you again, but we would really like to see this merged. Any news or additional comments on this?

embik commented:
I've pushed an update to documentation that marks this as being included in 1.90.0 (since 1.89.0 got released without merging it).

xiaozhu36 commented on website/docs/r/cs_kubernetes.html.markdown:
```diff
@@ -97,6 +97,9 @@ The following arguments are supported:
 * `node_name_mode` - (Optional, Available in 1.88.0+) Each node name consists of a prefix, an IP substring, and a suffix. For example, if the node IP address is 192.168.0.55, the prefix is aliyun.com, IP substring length is 5, and the suffix is test, the node name will be aliyun.com00055test. 
+* `service_account_issuer` - (Optional, ForceNew, Available in 1.90.0+) The issuer of the Service Account token for [Service Account Token Volume Projection](https://www.alibabacloud.com/help/doc-detail/160384.htm), corresponds to the `iss` field in the token payload. Set this to `"kubernetes.default.svc"` to enable the Token Volume Projection feature (requires specifying `api_audiences` as well) ...
+* `api_audiences` - (Optional, ForceNew, Available in 1.90.0+) A list of API audiences for [Service Account Token Volume Projection](https://www.alibabacloud.com/help/doc-detail/160384.htm). Set this to `["kubernetes.default.svc"]` if you want to enable the Token Volume Projection feature (requires specifying `service_account_issuer` as well.
+>>>>>>> 4bee3567... adjust documentation to include version availability
```
I think there is a conflict message. Please check it.

embik replied:
My bad! Good catch.

xiaozhu36 commented on website/docs/r/cs_kubernetes.html.markdown:
```diff
@@ -97,6 +97,9 @@ The following arguments are supported:
 * `user_data` - (Optional, Available in 1.81.0+) Windows instances support batch and PowerShell scripts. If your script file is larger than 1 KB, we recommend that you upload the script to Object Storage Service (OSS) and pull it through the internal endpoint of your OSS bucket.
 * `exclude_autoscaler_nodes` - (Optional, Available in 1.88.0+) Exclude autoscaler nodes from `worker_nodes`. default: false 
 * `node_name_mode` - (Optional, Available in 1.88.0+) Each node name consists of a prefix, an IP substring, and a suffix. For example, if the node IP address is 192.168.0.55, the prefix is aliyun.com, IP substring length is 5, and the suffix is test, the node name will be aliyun.com00055test. 
+* `service_account_issuer` - (Optional, ForceNew, Available in 1.90.0+) The issuer of the Service Account token for [Service Account Token Volume Projection](https://www.alibabacloud.com/help/doc-detail/160384.htm), corresponds to the `iss` field in the token payload. Set this to `"kubernetes.default.svc"` to enable the Token Volume Projection feature (requires specifying `api_audiences` as well) ...
```
The next version is 1.91.0 and please update it.

embik replied:
Done!

embik commented:
I've addressed the feedback and updated the code accordingly.

embik commented:
I've once again rebased this onto the most recent master and included version notes for a potential upcoming release.

demonwy commented:
@embik  thanks for your hard work.
